src/utils.js

```diff
@@ -79,16 +79,19 @@ module.exports = {
 		return new Promise((resolve, reject) => {
 			function getResponse() {
 				// Present the user with a prompt for input.
-				res.reply(prompt);
+				robot.emit('ibmcloud.formatter', { response: res, message: prompt});
+
 
 				// Control the response when the timeout expires.
 				dialog.dialogTimeout = function(msg) {
 					/* istanbul ignore next */
-					res.reply(i18n.__('conversation.timed.out'));
+					robot.emit('ibmcloud.formatter', { response: res, message: i18n.__('conversation.timed.out')});
 				};
 
 				// Handle the expected response.
 				dialog.addChoice(regex, (msg) => {
+					msg.match[1] = that.stripBotName(robot.name, msg.match[1]);
```

Comment:
Ah yeah the dreaded direct message issue. Should we also remove it from `match[0]` as well then?